src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy

```diff
@@ -270,6 +270,12 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
         buildFile << """
             task verifyImageId {
                 dependsOn buildImage
+
+                doLast {
+                    // Expose issues with getImageId returning null
+                    // https://github.com/bmuschko/gradle-docker-plugin/issues/818
+                    buildImage.getImageId().get()
```

Comment:
We'll need to assert that the value is not null. We should also have this check in the different test cases at every step along the way.

Reply:
@bmuschko Would you have any additional feedback on how I implemented this validation across the workflow in the review feedback commit?